python/lsst/ip/isr/ampOffset.py

```diff
@@ -18,88 +18,433 @@
+        neighbors, sides = [], []
+        sideLookup = {
+            0: (r + 1, c),
+            1: (r, c + 1),
+            2: (r - 1, c),
+            3: (r, c - 1),
+        }
+        for side, (row, column) in sideLookup.items():
+            if 0 <= row < m and 0 <= column < n:
+                neighbors.append(ampIds[row][column])
+                sides.append(side)
+        return neighbors, sides
+
+    def getInterfaceOffset(self, ampIdA, ampIdB, edgeA, edgeB):
+        """Calculate the amp offset for a given interface between two
+        amplifiers.
 
         Parameters
         ----------
-        exposure : `lsst.afw.image.Exposure`
-            Exposure to be corrected for any amp offsets.
+        ampIdA : int
+            ID of the first amplifier.
+        ampIdB : int
+            ID of the second amplifier.
+        edgeA : numpy.ndarray
+            Amp edge for the first amplifier.
+        edgeB : numpy.ndarray
+            Amp edge for the second amplifier.
+
+        Returns
+        -------
+        interfaceOffset : float
+            The calculated amp offset value for the given interface between
+            amps A and B.
+        """
+        sctrl = StatisticsControl()
+        # NOTE: Taking the difference with the order below fixes the sign flip
+        # in the B matrix.
+        edgeDiff = edgeA - edgeB
+        if self.config.ampEdgeWindow == 0:
+            if len(edgeDiff) > self.shortAmpSide:
+                window = self.shortAmpSide
+            else:
+                window = int(self.shortAmpSide**2 / self.longAmpSide)
+            self.log.info(
+                f"ampEdgeWindow automatically set to {window} pixels, for the "
+                f"amp {ampIdA} interfacing amp {ampIdB}, as value of 0 or no "
+                "input is provided for ampEdgeWindow."
+            )
+        else:
+            window = self.config.ampEdgeWindow
+        # Compute rolling averages.
+        edgeDiffSum = np.convolve(np.nan_to_num(edgeDiff), np.ones(window), "same")
+        edgeDiffNum = np.convolve(~np.isnan(edgeDiff), np.ones(window), "same")
+        edgeDiffAvg = edgeDiffSum / np.clip(edgeDiffNum, 1, None)
+        edgeDiffAvg[np.isnan(edgeDiff)] = np.nan
+        # Take clipped mean of rolling average data as amp offset value.
+        interfaceOffset = makeStatistics(edgeDiffAvg, MEANCLIP, sctrl).getValue()
+        # Perform a couple of do-no-harm safety checks:
+        # a) The fraction of unmasked pixel rows is > ampEdgeMinFrac,
+        # b) The absolute offset ADU value is < ampEdgeMaxOffset.
+        ampEdgeGoodFrac = 1 - (np.sum(np.isnan(edgeDiffAvg)) / len(edgeDiffAvg))
+        minFracFail = ampEdgeGoodFrac < self.config.ampEdgeMinFrac
+        maxOffsetFail = np.abs(interfaceOffset) > self.config.ampEdgeMaxOffset
+        if minFracFail or maxOffsetFail:
+            interfaceOffset = 0
+        interfaceId = f"{ampIdA}{ampIdB}"
+        self.log.debug(
```

Comment:
I think this needs a message at a different level (warning?) if this fails.  Otherwise, no-offset cases and failed measurements result look identical.

Reply:
Done!